jobscheduler/scheduler-component-shared/src/java/org/sakaiproject/component/app/scheduler/jobs/cm/processor/sis/MembershipProcessor.java

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright (c) 2003 The Apereo Foundation
+ *
+ * Licensed under the Educational Community License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *             http://opensource.org/licenses/ecl2
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sakaiproject.component.app.scheduler.jobs.cm.processor.sis;
+
+import lombok.extern.slf4j.Slf4j;
+import org.sakaiproject.component.app.scheduler.jobs.cm.processor.ProcessorState;
+import org.sakaiproject.coursemanagement.api.EnrollmentSet;
+import org.sakaiproject.coursemanagement.api.Membership;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+@Slf4j
+public class MembershipProcessor extends AbstractCMProcessor {
+
+    private final static Map<String, Set<String>> getInstructorMap(final ProcessorState state) {
+        return (Map<String, Set<String>>) state.getConfiguration().get("instructorMap");
+    }
+
+    private final static Map<String, List<String[]>> getCourseSetMembershipMap(final ProcessorState state) {
+        return (Map<String, List<String[]>>) state.getConfiguration().get("courseSetMembersMap");
+    }
+
+    private final static Map<String, List<String[]>> getMembersMap(final ProcessorState state) {
+        return (Map<String, List<String[]>>) state.getConfiguration().get("membersMap");
+    }
+
+    public ProcessorState init(Map<String, Object> config) {
+        ProcessorState state = super.init(config);
+
+        state.getConfiguration().put("instructorMap", new HashMap<String, Set<String>>());
+        state.getConfiguration().put("membersMap", new HashMap<String, List<String[]>>());
+        state.getConfiguration().put("courseSetMembersMap", new HashMap<String, List<String[]>>());
+
+        return state;
+    }
+
+    public void processRow(String[] data, ProcessorState state) throws Exception {
+        Map<String, Set<String>> instructorMap = getInstructorMap(state);
+        Map<String, List<String[]>> membersMap = getMembersMap(state);
+        Map<String, List<String[]>> courseSetMembersMap = getCourseSetMembershipMap(state);
+
+        String mode = data[0];
+
+        if (data[0].equalsIgnoreCase("instructor")) {
```

Comment:
I would flip the operands here for null safety:

`if ("instructor".equalsIgnoreCase(data[0]))`